### ISSUE TITLE
multi-line axis titles unable to push margins

### ISSUE THREAD

**Issue by laura-cbre** [plotly/plotly.js#4299]
Automargin updates margin size correctly if axis title is large, due to say using a large font.
However, it does not update margin size if axis title is large due to line breaks.

` fig.update_xaxes({'title_text': 'hiya', 'title_font': {'size': 100}, 'automargin': True})`
![image](https://user-images.githubusercontent.com/56594408/67476818-b3733800-f650-11e9-9648-72479f61c2de.png)

`fig.update_xaxes({'title_text': 'h<br>i<br>y<br>a', 'title_font': {'size': 100}, 'automargin': True})`
![image](https://user-images.githubusercontent.com/56594408/67476898-d7367e00-f650-11e9-9124-07da1af91976.png)

Their data attributes:
```python
(Scatter({
     'hoverinfo': 'x+name+text',
     'line': {'color': '#e53935'},
     'marker': {'color': '#e53935', 'line': {'color': '#e53935'}},
     'mode': 'lines',
     'name': 'a',
     'text': [1.000, 2.000, 3.000, 4.000],
     'x': array([datetime.datetime(2018, 1, 1, 0, 0),
                 datetime.datetime(2018, 1, 5, 0, 0),
                 datetime.datetime(2018, 1, 8, 0, 0),
                 datetime.datetime(2018, 1, 15, 0, 0)], dtype=object),
     'y': array([1, 2, 3, 4])
 }),)
```
First chart's layout attribute
```python
Layout({
    'title': {'text': ''},
    'xaxis': {'automargin': True, 'title': {'font': {'size': 100}, 'text': 'hiya'}},
    'yaxis': {'rangemode': 'normal', 'title': {'text': 'degC'}}
})
```
Second chart's layout attribute

```python
Layout({
    'title': {'text': ''},
    'xaxis': {'automargin': True, 'title': {'font': {'size': 100}, 'text': 'h<br>i<br>y<br>a'}},
    'yaxis': {'rangemode': 'normal', 'title': {'text': 'degC'}}
})
```

**Comment by etpinard** [comment]
Multi-line axis titles are in fact considered in the axis automargin computations (see example: https://codepen.io/etpinard/pen/MWWmoKV). For those interested, the logic is here:

https://github.com/plotly/plotly.js/blob/99fc02f50cb7e10a616b1c4928e5f7e16809c57c/src/plots/cartesian/axes.js#L2718-L2730

The problem with your graph is that the automargin push caused by your very large (multi-line) axis title is too big and simply dropped. The logic is here:

https://github.com/plotly/plotly.js/blob/99fc02f50cb7e10a616b1c4928e5f7e16809c57c/src/plots/plots.js#L1842-L1851

In other words, having your axis title push the margin would leave almost no room for the subplot, so we drop it.

I understand the behaviour is debatable (more info in https://github.com/plotly/plotly.js/issues/771#issuecomment-522146774), but we decided not to change it in the `v1.x` series. We might consider improvements for upcoming major releases.

**Comment by laura-cbre** [comment]
Hi etpinard thanks for your reply.

In that case the multi-line maximum size is smaller than the single line maximum size as shown below:

`sc.chart_fig.update_xaxes({'title_text': 'hiya', 'title_font': {'size': 230}, 'automargin': True})`
![image](https://user-images.githubusercontent.com/56594408/67505598-7d9b7700-f683-11e9-9ed5-e6635018b192.png)

Automargin stops working for just 5 lines, at font size 12
`sc.chart_fig.update_xaxes({'title_text': 'h<br>i<br>y<br>a<br>a','title_font': {'size': 12}, 'automargin': True})`
![image](https://user-images.githubusercontent.com/56594408/67505628-8a1fcf80-f683-11e9-860e-ef3d57145bcb.png)

Is there a way of asking if its a subplot, and if not, not impose a maximum size?

The reason I'm asking is there is no chart caption functionality. So I need to either use the xaxis title or annotations as a workaround for adding a chart caption. Ideally I'd be able to add up to 20 lines.

The layout objects of the above charts:
```python
Layout({
    'title': {'text': ''},
    'xaxis': {'automargin': True, 'title': {'font': {'size': 230}, 'text': 'hiya'}},
    'yaxis': {'rangemode': 'normal', 'title': {'text': 'degC'}}
})
```

```python
Layout({
    'title': {'text': ''},
    'xaxis': {'automargin': True, 'title': {'font': {'size': 12}, 'text': 'h<br>i<br>y<br>a<br>a'}},
    'yaxis': {'rangemode': 'normal', 'title': {'text': 'degC'}}
})
```

**Comment by etpinard** [comment]
> Automargin stops working for just 5 lines, at font size 12

In general, automargin will "stop working" when the push value is greater than half of the graph's height.

> Is there a way of asking if its a subplot, and if not, not impose a maximum size?

I'd recommend setting the bottom margin yourself using `layout.margin.b`

**Comment by laura-cbre** [comment]
> I'd recommend setting the bottom margin yourself using layout.margin.b

I will have many charts, each with a different number of lines, so I will have to write some hacky code to increase `layout.margin.b` as some function of the number of lines. I was just hoping I could avoid this.

**Comment by archmoj** [comment]
The problem appear to be related to computing the size of multi-line text.
Here is a [minimal codepen](https://codepen.io/MojtabaSamimi/pen/abmdOKb?editors=0010).

**Comment by gvwilson** [comment]
Hi - we are trying to tidy up the stale issues and PRs in Plotly's public repositories so that we can focus on things that are still important to our community. Since this one has been sitting for several years, I'm going to close it; if it is still a concern, please add a comment letting us know what recent version of our software you've checked it with so that I can reopen it and add it to our backlog. Thanks for your help - @gvwilson
